rfc/0008-builtin-blocktypes/README.md

```diff
@@ -0,0 +1,209 @@
+
+SPDX-License-Identifier: AGPL-3.0-only
+-->
+
+# RFC 0008: Builtin block types
+
+|             |                      |
+|-------------|----------------------|
+| Feature Tag | `builtin-blocktypes` |
+| Status      | `DRAFT`              | <!-- Possible values: DRAFT, DISCUSSION, ACCEPTED, REJECTED -->
+| Responsible | `@felix-oq`          |
+<!-- 
+  Status Overview:
+  - DRAFT: The RFC is not ready for a review and currently under change. Feel free to already ask for feedback on the structure and contents at this stage.
+  - DISCUSSION: The RFC is open for discussion. Usually, we open a PR to trigger discussions.
+  - ACCEPTED: The RFC was accepted. Create issues to prepare implementation of the RFC.
+  - REJECTED: The RFC was rejected. If another revision emerges, switch to status DRAFT.
+-->
+
+## Summary
+
+TODO
+
+## Motivation
+
+Block types only exist implicitly.
+Support declaration of blocktypes that are built into the language, so users can look up IO types and properties.
+No semantic changes but can serve as a foundation for composite block types.
+
+## Explanation
+
+- `builtin` keyword to indicate that the blocktype is built into the language
+  - Produces an error if the blocktype is unknown
+- `input` / `output` keywords to type the input and output of the blocktype
+  - If omitted, a blocktype has no input or output
+- `property` keyword to define a property with a name and a type
+  - Optionally, a default value can be assigned
+- Adds new keywords for existing types:
+  - Valuetypes: `regex`, `cellrange` / `row` / `column` / `cell`, `valuetype-assignment`
+  - Typed collections: `collection<type>`
+    - Nested collections are not supported for now
+  - IO types: `File`, `FileSystem`, `TextFile`, `Sheet`, `Table`
+
+### Syntax for current block types
+
+<details>
+<summary>Click to show</summary>
+
+```jayvee
+builtin blocktype HttpExtractor {
+    property url oftype text;
+
+    output oftype File;
+}
+
+builtin blocktype ArchiveInterpreter {
+    input oftype File;
+
+    property archiveType oftype text;
+
+    output oftype FileSystem;
+}
+
+builtin blocktype FilePicker {
+    input oftype FileSystem;
+
+    property path oftype text;
+
+    output oftype File;
+}
+
+builtin blocktype TextFileInterpreter {
+    input oftype File;
+
+    property encoding oftype text;
+    property lineBreak oftype regex;
+
+    output oftype TextFile;
+}
+
+builtin blocktype TextLineDeleter {
+    input oftype TextFile;
+
+    property lines oftype collection<integer>;
+
+    output oftype TextFile;
+}
+
+builtin blocktype TextRangeSelector {
+    input oftype TextFile;
+
+    property lineFrom oftype integer;
+    property lineTo oftype integer;
+
+    output oftype TextFile;
+}
+
+builtin blocktype CSVInterpreter {
+    input oftype TextFile;
+
+    property delimiter oftype text: ",";
+    property enclosing oftype text: "";
+    property enclosingEscape oftype text: "";
```

Comment:
Currently, we could simplify like that. My guts feeling says that it could hamper future changes, though. I'd keep pipeline and blocks as separate concepts.

However, there might be a relation between a `property` (compile-time parameter) and a `parameter` (runtime parameter).

Reply:
Not sure what I tried to say here... My best guess: `property` and `parameter` should follow a similar syntax (even though they are not the same thing)